### ISSUE TITLE
Wreckless - The Yakuza Missions - SoftLock

### ISSUE THREAD

**Issue by theboy181** [xemu-project/xemu#1129]
### Title

https://xemu.app/titles/4156000a/#Wreckless-The-Yakuza-Missions
AV-010 [4156000a]

### Bug Description

Game will softlock when entering the game and, or menus.  Looking at the  _info_ registers_ and the _games code_ it was a loop in the audio that locks up during after the test is called.

0x0012cd7b:  testw  $0x8000,0x12(%ecx)
 

### Expected Behavior

Game should not lock up.

### xemu Version

Version:      0.7.55
Branch:       master
Commit:       db389b15086753cb7b72e05b24af205baa742bc3
Date:         Sun Jun 26 05:58:57 UTC 2022

### System Information

CPU:          12th Gen Intel(R) Core(TM) i9-12900KF
OS Platform:  Windows
OS Version:   Windows
Manufacturer: NVIDIA Corporation
GPU Model:    NVIDIA GeForce RTX 3090/PCIe/SSE2
Driver:       4.0.0 NVIDIA 512.59
Shader:       4.00 NVIDIA via Cg compiler

### Additional Context

 As a workaround you can apply this patch to the xbe to get the game working.
 
Audio lock fix (find/replace hex)
FF FF 5E C9 C3 F6 41 12 01 74 08 66 F7 41 12 00 = FF FF 5E C9 C3 90 90 90 90 74 08 66 F7 41 12 00

**Comment by Triticum0** [comment]
Other games affected:
[~~Ultra Bust-A-Move~~](https://xemu.app/titles/4d4a000c#Ultra-Bust-A-Move) #822 #579
[~~Outlaw Golf~~](https://xemu.app/titles/544d0006#Outlaw-Golf) #579
[~~Outlaw Golf: 9 More Holes of X-Mas~~](https://xemu.app/titles/53530006#Outlaw-Golf-9-More-Holes-of-X-Mas)
[~~Kikou Heidan J-Phoenix +~~](https://xemu.app/titles/54410001/#Kikou-Heidan-J-Phoenix)

**Comment by theboy181** [comment]
Where are these games hanging, and how do you know its audio loop? can you pinpoint where the issue is? I would like to take a look. 
@Triticum0

**Comment by theboy181** [comment]
> Other games affected: [Ultra Bust-A-Move](https://xemu.app/titles/4d4a000c#Ultra-Bust-A-Move) #822 #579 [Outlaw Golf](https://xemu.app/titles/544d0006#Outlaw-Golf) #579 [Outlaw Golf: 9 More Holes of X-Mas](https://xemu.app/titles/53530006#Outlaw-Golf-9-More-Holes-of-X-Mas) [Kikou Heidan J-Phoenix +](https://xemu.app/titles/54410001/#Kikou-Heidan-J-Phoenix)

Was unable to test the last game on this list, I don't think they are related to the softlock. Bust a move does has an issue loading into the games menu, but doesnt lock up, and I was unable to get the game to softlock in Outlaw Golf series.

**Comment by Triticum0** [comment]
Yeah your right, It was a while back when I last tested them. Sorry for the trouble.

**Comment by Triticum0** [comment]
I would redo the report for outlaw golf if it's fixed. Note: only affect [Outlaw Golf: 9 More Holes of X-Mas](https://xemu.app/titles/53530006#Outlaw-Golf-9-More-Holes-of-X-Mas) none of the others in the series

**Comment by Triticum0** [comment]
@theboy181 Yeah, I think you're right with the games I said originally are unrelated. Sorry about that but https://xemu.app/titles/45530001/#Mad-Dash-Racing Is properly related look at the report.

**Comment by Triticum0** [comment]
https://xemu.app/titles/45410083/#Black
https://xemu.app/titles/4a410002/#Night-Caster-II-Equinox

**Comment by jorfgm** [comment]
Work on my PC, my setting are Memory 128mb , AV PACK SCART, HARDFPU emulation enabled. 
My PC, Ryzen 5 5600, 16gb ram ddr4 3200mhz, 1tb ssd adata, gtx 1060 6gb..
rom tested from archive(org)

**Comment by jorfgm** [comment]
> 

If I change my settings the game freezes in the menu.
With these settings the game does not work.

It freezes after choosing the level to play.
memory 64mb
AV Pack Scart (any other option)
HARD FPU Enable

It freezes as soon as it starts
memory 64mb
AV PACK Scart (any other option)
Hard FPU disable

It freezes as soon as you press a button.
Memory 128mb
AV PACK SCART (any other option)
Hard FPU Disable

**Comment by darkfalzx** [comment]
A weird observation:
While cold-booting the game will always lead to the same freeze on the difficulty select screen, starting any other game (I tried Tork, Castlevania CoD, Dino Crisis 3 and Burnout 3), saving or creating data in it (could be just menu options), THEN starting Wreckless may *SOMETIMES* get the game to pass the difficulty selection screen and get into the game-proper.
Retrying the mission from the pause menu works fine.
The game runs well, though with a few visual glitches (like incorrect drawing depth on some scenery and partially broken shadow maps).
Completing mission and starting another will always cause the freeze.
Restarting Wrickless will always cause the freeze.
*SOMETIMES* You can even restart the emulator between running another game and Wreckless - only as long as any game besides Wreckless wrote to the HDD image previously!

Maybe there is something to it, but maybe I'm just seeing a pattern where there is none.

**Comment by theboy181** [comment]
I got the game booting with hacks over a year ago. However there seems to be no appetite for the devs to fix this game. Too bad as the game is still really good to look at.

**Comment by darkfalzx** [comment]
Please don't leave me hanging - where can I find said hacks?

**Comment by theboy181** [comment]
I would take time to read this thread.

**Comment by Soxbrother** [comment]
With the latest Xemu 0.7.118 :
-Before the sound hex fix, it would freeze on the difficulty mode screen.
-After the hex fix, it goes beyond the difficulty screen and starts the game, but only black screen with sound.

So I then tried Xemu 0.7.55, because that's the version the fix was tested with :
-Bingo, game is playable, no more black screen !!

Thanks @theboy181  for the fix

Probably a version closer to the latest version is also playable, but that needs to be tested.
Somewhere along the way, the game went from playable to unplayable with the latest version.

**Comment by Triticum0** [comment]
The game more stable now I can get in-game some times but crash occurs after finishing the first level. v0.8.47

**Comment by LIFTERxPULLER** [comment]
I appreciate the advice with the Hex-edit @theboy181, it did allow me to finally get into the game after all these years of trying to emulate it properly. For some odd reason, my game's crashing at the end of each mission; it simply loops the "mission complete" dialogue until I close the app. I may try another version of the emulator per another post in this thread, but just curious if we know any other fix for it.

**Comment by jyetcv** [comment]
Is this still relevant? I played the game on xemu 0.8.111 recently without any issue.

**Comment by Triticum0** [comment]
Complete the game then I say probably fixed. crashed for me after first mission so will retest to see if it any better. Performance isn't good enough to be playable anyway so doesn't really matter if the crashing fix or not.

**Comment by jyetcv** [comment]
This game used to not work at all for me, but with the latest xemu I have completed the game.

The only bug I encountered was the one in this video, and I think it was because I completed the mission exactly as the timer ran out, so the game got confused:

https://github.com/user-attachments/assets/82d932a8-654f-4503-b9fd-d539124bc3c2

**Comment by Triticum0** [comment]
Let @theboy181 decide I tested 2 levels and It didn't crash. but that video does look like a crash so I'm torn. performance does seem alot better on linux minus big shader stutters.
